dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -480,9 +480,9 @@ public CloseableClient build() {
                             new InstrumentedHostnameVerifier(
                                     new DefaultHostnameVerifier(), name, conf.taggedMetricRegistry())) {
                         @Override
-                        public Socket createSocket(final HttpContext context) throws IOException {
+                        public Socket createSocket(final HttpContext _context) {
                             return socksProxyAddress == null
-                                    ? super.createSocket(context)
+                                    ? new Socket(Proxy.NO_PROXY)
```

Comment:
shouldn't this use `rawSocketFactory` to create the socket?

Reply:
It is, in a non-obvious way: The rawSocketFactory (SSLSocketFactory) is used with the vanilla socket created here, via [createSocket(Socket,String,int,boolean)](https://github.com/openjdk/jdk/blob/573eaceca559a8a0832b1e1a7181b2f21d3978c7/src/java.base/share/classes/javax/net/ssl/SSLSocketFactory.java#L281-L282).
Magic happens [here](https://github.com/apache/httpcomponents-client/blob/c395aad5adb4e9ee312559937c08382ea8215e7c/httpclient5/src/main/java/org/apache/hc/client5/http/ssl/SSLConnectionSocketFactory.java#L270-L274)

Reply:
That sounds more correct. I don't think it matches the previous behaviour which wouldn't have been using that one ([here](https://github.com/apache/httpcomponents-client/blob/47d0946372e757ca6e7a366e340f0dba6686b012/httpclient5/src/main/java/org/apache/hc/client5/http/ssl/SSLConnectionSocketFactory.java#L194) using a default one, unless the config one is set to the default?), but sounds like it should have been. I'll change it

Reply:
Ahh, the naming confused me, so this is creating the actual raw socket, that will then be wrapped with TLS